packages/web3-utils/src/errors.ts

```diff
@@ -155,3 +154,33 @@ export class InvalidTypeAbiInput extends Web3Error {
 		super(value, 'components found but type is not tuple');
 	}
 }
+
+// TODO Type of value for all errors should be any as
+// any value can be passed when an error occurs
+export class InvalidBlockNumberOrTag extends Web3Error {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	public constructor(value: any) {
+		super(value, 'invalid block number or tag given');
+	}
+}
+
+export class InvalidHexString8Bytes extends Web3Error {
+	public constructor(value: string) {
+		super(value, 'invalid 8 byte hex string given');
+	}
+}
+
+export class InvalidHexString32Bytes extends Web3Error {
```

Comment:
I would sugges to create one generic constructor for such cases. e.g. `InvalidHexString` would be sufficient to cover all type of bytes data. We can provide more specific information in the error message itself.